src/types/component.go

```diff
@@ -107,13 +110,48 @@ type ZarfComponentScripts struct {
 	After          []string `json:"after,omitempty" jsonschema:"description=Scripts to run after the component successfully deploys"`
 }
 
-// ZarfContainerTarget defines the destination info for a ZarfData target.
+// ZarfComponentActions are actionsets that map to different zarf package operations
+type ZarfComponentActions struct {
+	OnCreate ZarfComponentActionSet `json:"onCreate,omitempty" jsonschema:"description=Actions to run during package creation"`
```

Comment:
A random thought that I'm not sure we've considered yet..

At the moment we batch up all the images that the components declare so that they are all pulled at the same time during the later stage of package creation. Depending on how specific someone is trying to be with a ComponentAction.OnCreate command that might cause a potential issue b/c that isn't intuitive unless you've looked through the code.

Reply:
I think that's a good point, we moved to a single image collection for binary efficiencies (reuse of layer refs in a single tarball), but iirc it wasn't significant. It probably would make more sense to do this per component if it's not a major loss in efficiency.

Reply:
Agreed. 

I think this can be done in a separate PR so that we don't grow this one to be too big.